sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -985,7 +993,7 @@ def _validate_input(self, X, y, incremental):
                                  " `self.classes_` has %s. 'y' has %s." %
                                  (self.classes_, classes))
 
-        y = self._label_binarizer.transform(y)
+        y = self._label_binarizer.transform(y).astype(np.bool)
```

Comment:
I added this because the call to `_label_binarizer.transform` returns `int64`, which was upcasting other network parameters to `float64`  during `_backprop`.

Reply:
Let's include this explanation as a comment in the code. (This downcast to bool is to prevent upcasting when working with float32 data)